src/app/server/Server.h

```diff
@@ -90,7 +91,7 @@ class Server
     static Server & GetInstance() { return sServer; }
 
 private:
-    Server() : mCommissioningWindowManager(this) {}
+    Server() : mCommissioningWindowManager(this), mGroupsProvider(mServerStorage) {}
```

Comment:
This couples server storage to GroupsDataProvider: in other words, any GroupDataProvider *must* use this implementation of storage.

This makes the usage of an interface here useless since we depend in the common server.cpp on a *specific* impl that cannot be overriden (e.g. GroupDataProviderImpl). The GroupDataProvider impl should be injected into Server, and so should the storage impl, rather than belonging to this common code